client/state/utils.js

```diff
@@ -25,6 +25,22 @@ export function isValidStateWithSchema( state, schema, checkForCycles = false, b
 }
 
 /**
+ * Given an action object or thunk, returns an updated object or thunk which
+ * will include additional meta in the action (as provided) when dispatched.
+ *
+ * @param  {(Function|Object)} action Action object or thunk
+ * @param  {Object}            meta   Additional meta to include in action
+ * @return {(Function|Object)}        Augmented action object or thunk
+ */
+export function dispatchWithMeta( action, meta ) {
+	if ( 'function' !== typeof action ) {
+		return { ...action, meta };
```

Comment:
We're overwriting any existing meta here. If, for example, we were to extend an action with meta already, such as some event wrapped by the analytics calls, this would eliminate those calls by overwriting the entire property. We can do this without destroying existing data:

```js
return { ...action, meta: { ...get( action, 'meta', {} ), ...meta } };
```

Er…I think that's right.